### PR TITLE
Enable PPA builds

### LINKED ISSUE #880: Add PPA for Dash?
Seems this was discussed here: https://www.dash.org/forum/threads/ubuntu-repository.1670/

Would make installation / upgrades much easier for users on Linux.

### PR DESCRIPTION
**Pull request by schinzelh**
- this merges the latest Bitcoin changes and should enable us to provide Dash PPAs via https://launchpad.net/~dash.org/+archive/ubuntu/dash

Closes #880

### CODE REVIEW

File: contrib/debian/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ To install:
 
 If you build yourself, you will either need to modify the paths in
 the .desktop file or copy or symlink your dash-qt binary to `/usr/bin`
-and the `../../share/pixmaps/dash128.png` to `/usr/share/pixmaps`
+and the `../../share/pixmaps/bitcoin128.png` to `/usr/share/pixmaps`
```
Comment (UdjinM6) [inline contrib/debian/README.md]: Actually, if we want to avoid conflicts with bitcoin packages imo it's time to fix bitcoin->dash for pixmaps https://github.com/dashpay/dash/tree/master/share/pixmaps (and references to them)

File: contrib/debian/changelog [hunk 2]
```diff
@@ -179,7 +241,7 @@ bitcoin (0.5.3-natty0) natty; urgency=low
 bitcoin (0.5.2-natty1) natty; urgency=low
 
   * Remove mentions on anonymity in package descriptions and manpage.
-    These should never have been there, bitcoin isn't anonymous without
+    These should never have been there, bitcoin isnt anonymous without
```
Comment (UdjinM6) [inline contrib/debian/changelog]: changes starting from this line look more like a rollback, bringing errors/misspellings...

File: contrib/debian/control [hunk 3]
```diff
@@ -1,41 +1,45 @@
...
 Standards-Version: 3.9.2
-Homepage: https://www.dash.org/
+Homepage: https://dash.org/
 Vcs-Git: git://github.com/dashpay/dash.git
 Vcs-Browser: https://github.com/dashpay/dash
 
 Package: dashd
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}
 Description: peer-to-peer network based digital currency - daemon
```
Comment (UdjinM6) [inline contrib/debian/control]: again - more like a rollback

File: contrib/debian/manpages/dashd.1 [hunk 4]
```diff
@@ -1,8 +1,8 @@
-.TH DASHD "1" "February 2016" "dashd 0.12"
+.TH DASHD "1" "June 2016" "dashd 0.12"
 .SH NAME
-Dash \- peer-to-peer network based digital currency
+dashd \- peer-to-peer network based digital currency
 .SH SYNOPSIS
-dashd [options] <command> [params]
+dashd [options] <command> [params]
```
Comment (UdjinM6) [inline contrib/debian/manpages/dashd.1]: trailing space

File: contrib/debian/rules [hunk 5]
```diff
@@ -18,7 +18,7 @@ override_dh_auto_clean:
 # Yea, autogen should be run on the source archive, but I like doing git archive
 override_dh_auto_configure:
 	./autogen.sh
-	./configure
+	./configure --without-miniupnpc --with-gui=qt4
```
Comment (UdjinM6) [inline contrib/debian/rules]: `--without-miniupnpc` makes connectivity significantly worse for those who are behind routers afaik, is there a reason to configure like this? also why qt4?

File: contrib/debian/changelog (hunk 2 above)
Comment (schinzelh) [inline contrib/debian/changelog]: Yep, it's a rollback - you don't change changelogs afterwards

File: contrib/debian/control (hunk 3 above)
Comment (schinzelh) [inline contrib/debian/control]: I prefer dash.org instead of www.dash.org :)

File: contrib/debian/rules (hunk 5 above)
Comment (schinzelh) [inline contrib/debian/rules]: That's what Bitcoin is doing - i guess for compatibilty with 5 different Ubuntu versions...

Comment (schinzelh) [inline contrib/debian/rules]: https://launchpad.net/~bitcoin/+archive/ubuntu/bitcoin/+files/bitcoin_0.12.1-trusty1.debian.tar.gz

**Comment by schinzelh** [comment]
Btw, this is just for documentation purposes, Bitcoin is not even maintaining their contrib/debian folder anymore

File: contrib/debian/changelog (hunk 2 above)
Comment (UdjinM6) [inline contrib/debian/changelog]: ah, ok, good point :)

File: contrib/debian/rules (hunk 5 above)
Comment (UdjinM6) [inline contrib/debian/rules]: hmm, ok

File: contrib/debian/README.md (hunk 1 above)
Comment (schinzelh) [inline contrib/debian/README.md]: This is out of scope of this PR - i just wanted it to be accurate (and reflect the current state) :)

File: contrib/debian/control (hunk 3 above)
Comment (UdjinM6) [inline contrib/debian/control]: well, that's questionable :) because dash.org redirects to www.dash.org as an actual Homepage and dash.org is also used for a bunch of other things like `forum.`, `explorer.` etc

File: contrib/debian/rules (hunk 5 above)
Comment (schinzelh) [inline contrib/debian/rules]: I need to test it on all 5 versions anyway, if something does not work out we'll adapt

Comment (schinzelh) [inline contrib/debian/rules]: Re. miniupnp: As far as i know Bitcoin disabled it because of security concerns

File: contrib/debian/control (hunk 3 above)
Comment (schinzelh) [inline contrib/debian/control]: OK, convinced, will push a change :)

File: contrib/debian/manpages/dashd.1 (hunk 4 above)
Comment (schinzelh) [inline contrib/debian/manpages/dashd.1]: Agreed

File: contrib/debian/control (hunk 3 above)
Comment (UdjinM6) [inline contrib/debian/control]: also description below
> Dash is ...

was matching readme and a newly proposed version doesn't. I like readme version more then a newly proposed one, imo current one is more accurate, no need to change it.

**Comment by schinzelh** [comment]
Pushed 7ad1ea8

File: contrib/debian/README.md (hunk 1 above)
Comment (UdjinM6) [inline contrib/debian/README.md]: Errr... I'm not sure if it makes sense to enable ppa builds which will conflict with some existing one... how about fixing /contrib/debian/dash-qt.install instead (at least that's the only reference I found) and renaming bitcoin128.png to dash128.png?

Comment (UdjinM6) [inline contrib/debian/README.md]: actually there are 2 more files referenced in /contrib/debian/dash-qt.install - bitcoin32.xpm and bitcoin16.xpm and I can't find any more references to them so renaming all 3 of them shouldn't break anything else imo

Comment (schinzelh) [inline contrib/debian/README.md]: Okok :) So we'll rename all bitcoin*.* files in share/pixmap and their references?

image

Comment (UdjinM6) [inline contrib/debian/README.md]: yep, will require to fix winfows packaging though because of bitcoin.ico (Makefile.am and setup.nsi.in)... or let's start with bitcoin*.png and bitcoin*.xpm for debian packaging only changes :)

**Comment by schinzelh** [comment]
Pushed 05e0a1240

**Comment by schinzelh** [comment]
Pushed da7ef9c - rename the missing `bitcoin*.png` and `bitcoin*.xpm` to their dash counterparts

File: contrib/debian/manpages/dashd.1 [hunk 6]
```diff
@@ -1,6 +1,6 @@
-.TH DASHD "1" "February 2016" "dashd 0.12"
+.TH DASHD "1" "June 2016" "dashd 0.12"
 .SH NAME
-Dash \- peer-to-peer network based digital currency
+dashd \- peer-to-peer network based digital currency
```
Comment (UdjinM6) [inline contrib/debian/manpages/dashd.1]: wrong find/replace :)

Comment (schinzelh) [inline contrib/debian/manpages/dashd.1]: You or me? :)

File: contrib/debian/control [hunk 7]
```diff
@@ -31,11 +32,14 @@ Package: dashd
...
- Dash is an experimental new digital currency that enables instant
- payments to anyone, anywhere in the world. Dash uses peer-to-peer
- technology to operate with no central authority: managing transactions
- and issuing money are carried out collectively by the network. Dash Core
- is the name of the open source software which enables the use of this currency.
+ Dash is a free open source peer-to-peer electronic cash system that
```
Comment (UdjinM6) [inline contrib/debian/control]: I probably commented on a wrong line previously, so rephrasing here...
This description was matching readme and a newly proposed version doesn't. I like readme version more then a newly proposed one, imo current one is more accurate, no need to change it. Also I'm not sure where is new one came from.

Comment (schinzelh) [inline contrib/debian/control]: I have no idea either. The one committed now is from Bitcoin 12.1 with Bitcoin replaced by Dash :)

File: contrib/debian/manpages/dashd.1 (hunk 6 above)
Comment (UdjinM6) [inline contrib/debian/manpages/dashd.1]: me :)

**Comment by schinzelh** [comment]
Pushed d0d2f0519b587459ba5e25c9cf678e104d38903e

**Comment by schinzelh** [comment]
Pushed 3dc2bab

**Comment by UdjinM6** [comment]
utACK 3dc2bab

Hope it actually works ;)
